demos/palm/python/docs-agent/scripts/populate_vector_database.py

```diff
@@ -0,0 +1,302 @@
+                    "md_hash": md_hash,
+                }
+                # Merges dictionaries with main metadata and additional metadata
+                metadata_dict_final = metadata_dict_main | metadata_dict_extra
+                str_uuid_file = str(uuid_file)
+                print("UUID: " + str_uuid_file)
+                print("Markdown hash: " + str(md_hash))
+                print("URL: " + url)
+                if toFile and toFile.strip():
+                    # Skip if the file size is larger than 10000 bytes (API limit)
+                    filesize = len(toFile)
+                    if filesize < 10000:
+                        if md_hash != "" and str_uuid_file != "":
+                            query = {}
+                            # The query looks for the UUID, which is unique and
+                            # compares to see if the hash has changed
+                            query = collection.get(
+                                include=["metadatas"],
+                                ids=str_uuid_file,
+                                where={"md_hash": {"$ne": md_hash}},
+                            )
+                            # Extract any id whose content may have changed
+                            id_to_remove = query["ids"]
+                            if id_to_remove != []:
+                                print("Out of date content.")
+                                # Delete the existing entry
+                                collection.delete(ids=id_to_remove)
+                                # Add a new entry
+                                collection.add(
+                                    documents=toFile,
+                                    metadatas=metadata_dict_final,
+                                    ids=str_uuid_file,
+                                )
+                                print("Updated.")
+                                updated_count += 1
+                                file_update = True
+                            else:
+                                query_2 = collection.get(
+                                    include=["metadatas"],
+                                    ids=str_uuid_file,
+                                    where={"md_hash": {"$eq": md_hash}},
+                                )
+                                id_up_to_date = query_2["ids"]
+                                if id_up_to_date != []:
+                                    print("Up to date content.")
+                                    unchanged_count += 1
+                                else:
+                                    collection.add(
+                                        documents=toFile,
+                                        metadatas=metadata_dict_final,
+                                        ids=str_uuid_file,
+                                    )
+                                    print("Added content.")
+                                    new_count +=1
```

Comment:
Could you run `black` over this? NBD but we can keep things consistent.

Reply:
Sure! Will run `black` again.

Reply:
Updated!